persistence-api/src/main/java/io/stargate/db/AuthenticatedUser.java

```diff
@@ -26,11 +26,25 @@
   @Nullable
   String token();
 
+  boolean usesTransitionalAuth();
+
   static AuthenticatedUser of(String userName) {
-    return ImmutableAuthenticatedUser.builder().name(userName).build();
+    return ImmutableAuthenticatedUser.builder().name(userName).usesTransitionalAuth(false).build();
```

Comment:
We are setting it to `false` for backward compatibility. Maybe extract it to a constant with `DEFAULT_USES_TRANSTIONAL_AUTH`?